daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -22,31 +22,42 @@ object PartialTransaction {
   final class Context private (
       val exeContext: Option[ExercisesContext], // empty if root context
       val children: BackStack[Value.NodeId],
-      protected val childrenSeeds: Stream[crypto.Hash]
+      protected val childrenSeeds: Int => Option[crypto.Hash]
   ) {
     def addChild(child: Value.NodeId): Context =
-      new Context(exeContext, children :+ child, childrenSeeds.drop(1))
-    def nextChildrenSeed: Option[crypto.Hash] = childrenSeeds.headOption
+      new Context(exeContext, children :+ child, childrenSeeds)
+    def nextChildrenSeed: Option[crypto.Hash] =
```

Comment:
As already before this PR, there is an assumption that `addChild` and `nextChildrenSeed` are always called in that order, right?

Reply:
First you build your node using `nextChildrenSeed` if this node require a seed (Create and Exercice) and then you add this node with `addChild`